lib/Onyx.js

```diff
@@ -614,7 +618,22 @@ function disconnect(connectionID, keyToRemoveFromEvictionBlocklist) {
  */
 // eslint-disable-next-line rulesdir/no-negated-variables
 function notifySubscribersOnNextTick(key, value) {
-    Promise.resolve().then(() => keyChanged(key, value));
```

Comment:
Probably the tests are failing because we are making the assumption in some places that as long as we wait for a promise then we should have the correct picture of the updated values. That works fine in the test environment since there's not much going on and as long as some kind of promise is waited on most of the ones we care about should also have run.

Since we are using a `setTimeout()` now we will have to deal with jest timers.

e.g. I see that the `cacheEvictionTest.js` test is failing. But if we put `jest.runAllTimers()` [here](https://github.com/Expensify/react-native-onyx/blob/0846e16fa97582c245fbdd966b37ce6d32e50465/tests/unit/cacheEvictionTest.js#L52) then it will pass.

```js
return Onyx.set(`${ONYX_KEYS.COLLECTION.TEST_KEY}${RECORD_TO_ADD}`, {test: 'add'})
    .then(() => {
        jest.runAllTimers();

        // Then our collection should no longer contain the evictable key
        expect(collection[`${ONYX_KEYS.COLLECTION.TEST_KEY}${RECORD_TO_EVICT}`]).toBe(undefined);
        expect(collection[`${ONYX_KEYS.COLLECTION.TEST_KEY}${RECORD_TO_ADD}`]).toStrictEqual({test: 'add'});
    });
```

However, I think it's probably even better to have the `setTimeout(runBatch, 0)` be a `Promise.resolve().then(runBatch)` - this change will fix all failing tests for me.

```js
function notifySubscribersOnNextTick(key, value) {
    pendingUpdates.push([key, value]);
    if (!scheduled) {
        scheduled = true;
        Promise.resolve().then(() => {
            scheduled = false;
            const updatesCopy = pendingUpdates;
            pendingUpdates = [];

            // eslint-disable-next-line prefer-arrow-callback
            unstable_batchedUpdates(function flushUpdatesOnyx() {
                for (let i = 0; i < updatesCopy.length; ++i) {
                    keyChanged(updatesCopy[i][0], updatesCopy[i][1]);
                }
            });
        });
    }
}
```